### PR TITLE
Adds support for checkbox question type

### LINKED ISSUE #27: Support Checkboxes as a new question type

### PR DESCRIPTION
**Pull request by dalmia**
Fixes #27  

## Summary
- updates calculations to account for checkbox as a new question type

## Test Plan

- [x] Schema file code updates are made for both BigQuery and Postgres
- [x] Tested locally
- [x] Tested on staging
- [ ] Tested on production

### CODE REVIEW

File: schema/bigquery/SessionAnswer.js [hunk 1]
```diff
@@ -103,10 +103,10 @@ cube(`AggregateSessionMetrics`, {
       ROW_NUMBER() OVER (ORDER BY plio_id, user_id) as id,
       plio_id,
       user_id,
-      COUNT(case when question_type = 'mcq' AND answer IS NULL THEN NULL when question_type = 'subjective' AND answer IS NULL then NULL else 1 end) AS num_answered,
+      COUNT(case when question_type IN ('mcq', 'checkbox') AND answer IS NULL THEN NULL when question_type = 'subjective' AND answer IS NULL then NULL else 1 end) AS num_answered,
```
Comment (deepansh96) [inline schema/bigquery/SessionAnswer.js]: Why not this?
```suggestion
      COUNT(case when question_type IN ('mcq', 'checkbox', 'subjective') AND answer IS NULL THEN NULL else 1 end) AS num_answered,
```

Comment (deepansh96) [reply schema/bigquery/SessionAnswer.js]: Also let's keep it consistent
SQL language strings should stay in uppercase
```suggestion
      COUNT(CASE WHEN question_type IN ('mcq', 'checkbox') AND answer IS NULL THEN NULL WHEN question_type = 'subjective' AND answer IS NULL THEN NULL ELSE 1 END) AS num_answered,
```

File: schema/bigquery/SessionAnswer.js [hunk 2]
```diff
@@ -103,10 +103,10 @@ cube(`AggregateSessionMetrics`, {
...
       COUNT(
         case
-          when question_type = 'mcq' AND answer = question_correct_answer then 1
+          when question_type IN ('mcq', 'checkbox') AND answer = question_correct_answer then 1
```
Comment (deepansh96) [inline schema/bigquery/SessionAnswer.js]: Here when doing `answer = question_correct_answer`, it automatically handles that the fact that the correct answer for `checkbox` type questions is an array?

Comment (deepansh96) [reply schema/bigquery/SessionAnswer.js]: Also, the same comment about staying consistent with the styles
Ideally we should do it in all the files

File: schema/postgres/Session.js [hunk 3]
```diff
@@ -79,7 +79,7 @@ cube(`GroupedSession`, {
         session.watch_time,
         session.retention,
         ROW_NUMBER() OVER(PARTITION BY session.user_id, session.plio_id
-                          ORDER BY session.watch_time DESC) AS rank
+                          ORDER BY session.id DESC) AS rank
```
Comment (deepansh96) [inline schema/postgres/Session.js:82]: Why were we picking up the highest watch_time waala session previously?
Now it makes sense that we should pick the latest session for a user-plio combination, but I'm not sure what was the reason for picking the session with highest watch time previously?

File: schema/postgres/SessionAnswer.js [hunk 4]
```diff
@@ -103,10 +103,10 @@ cube(`AggregateSessionMetrics`, {
       ROW_NUMBER() OVER (ORDER BY plio_id, user_id) as id,
       plio_id,
       user_id,
-      COUNT(case when question_type = 'mcq' AND answer IS NULL THEN NULL when question_type = 'subjective' AND answer IS NULL then NULL else 1 end) AS num_answered,
+      COUNT(case when question_type IN ('mcq', 'checkbox') AND answer IS NULL THEN NULL when question_type = 'subjective' AND answer IS NULL then NULL else 1 end) AS num_answered,
```
Comment (deepansh96) [inline schema/postgres/SessionAnswer.js]: Same comment as above
Why not this?
```suggestion
      COUNT(case when question_type IN ('mcq', 'checkbox', 'subjective) AND answer IS NULL else 1 end) AS num_answered,
```

**Review by deepansh96 — CHANGES_REQUESTED** [review]

File: schema/bigquery/SessionAnswer.js (hunk 2 above)
Comment (dalmia) [reply schema/bigquery/SessionAnswer.js]: > Here when doing answer = question_correct_answer, it automatically handles that the fact that the correct answer for checkbox type questions is an array?

yep

**Review by deepansh96 — APPROVED** [review]

**Pull request merged**